sdks/python/apache_beam/typehints/row_type.py

```diff
@@ -123,6 +123,12 @@ def from_fields(
         field_options=field_options,
         schema_registry=schema_registry)
 
+  def __call__(self, *args, **kwargs):
+    # We make RowTypeConstraint callable (defers to constructing the user type)
```

Comment:
Could you clarify why we need to make this a callable for Python to recognize it as a type (may be also update the comment here) ?

Reply:
I'm not sure if there's a better reference on this, but the CPython implementation explicitly considers anything that's callable a type:
https://github.com/python/cpython/blob/d348afa15d5a997e7a8e51c0f789f41cb15cc651/Lib/typing.py#L137-L167

Reply:
It looks like this is actually being relaxed in Python 3.11: https://github.com/python/cpython/commit/870b22b9c442d035190d2b8fb82256cd9a03da48#diff-ddb987fca5f5df0c9a2f5521ed687919d70bb3d64eaeb8021f98833a2a716887

Reply:
I see. Optionally, add a comment that this can be removed after we are fully in Python 3.11 (whenever that is).